src/string.rs

```diff
@@ -563,18 +660,41 @@ impl String {
     }
   }
 
-  // Get the ExternalStringResource for an external string.
-  //
-  // Returns None if is_external() doesn't return true.
+  /// Get the ExternalStringResource for an external string.
+  ///
+  /// Returns None if is_external() doesn't return true.
   pub fn get_external_string_resource(
     &self,
   ) -> Option<NonNull<ExternalStringResource>> {
     NonNull::new(unsafe { v8__String__GetExternalStringResource(self) })
   }
 
+  /// Get the ExternalOneByteStringResource for an external one-byte string.
+  ///
+  /// Returns None if is_external_onebyte() doesn't return true.
+  pub fn get_external_onebyte_string_resource(
+    &self,
+  ) -> Option<NonNull<ExternalOneByteStringResource>> {
+    if !self.is_external_onebyte() {
+      return None;
+    }
```

Comment:
nit: check not needed because `GetExternalStringResourceBase` returns null in this case anyway